File: examples/eApi v1.9/dotNET/CsobGTWDemo/CsobGatewayClientExample/Communication/DataObjects/EchoCustomerRequest.cs
using System.Text;
using Newtonsoft.Json;

namespace CsobGatewayClientExample.Communication.DataObjects;

public class EchoCustomerRequest : SignBaseRequest
{
    [JsonProperty("customerId", Required = Required.Always)]
    public string? CustomerId { get; set; }

    public override string ToSign()
    {
        var sb = new StringBuilder();
        Add(sb, MerchantId);
        Add(sb, CustomerId);
        Add(sb, Dttm);
        DeleteLast(sb);
        return sb.ToString();
    }
}

File: examples/eApi v1.9/java/eAPI-v19-connector/src/main/java/cz/monetplus/mips/eapi/v19/connector/entity/EchoCustomerRequest.java
package cz.monetplus.mips.eapi.v19.connector.entity;

import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NonNull;
import lombok.RequiredArgsConstructor;

@EqualsAndHashCode(callSuper = false)
@Data
@RequiredArgsConstructor
@JsonInclude(value = JsonInclude.Include.NON_NULL)
public class EchoCustomerRequest  extends SignBase {
    @NonNull
    private String merchantId;
    @NonNull
    private String customerId;


    @Override
    public String toSign() {
        StringBuilder sb = new StringBuilder();
        add(sb, merchantId);
        add(sb, customerId);
        add(sb, dttm);
        deleteLast(sb);
        return sb.toString();
    }
}
